### PR TITLE
PP-611 500 Error mapping Creating a Charge when Gateway account does not exist

### PR DESCRIPTION
**Pull request by rauligs**
- When Authorization succeed but Gateway account is Not Found in Connector, PublicAPI returns a 500 rather than a 400 Bad Request. This situation can't theoretically happen and a client can't recover from this without support.

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateChargeConnectorErrorResponseException.class);
+
+    @Override
+    public Response toResponse(CreateChargeConnectorErrorResponseException exception) {
+
+        int errorStatus = exception.getErrorStatus();
+
+        if (errorStatus == NOT_FOUND.getStatusCode()) {
+            LOGGER.warn("Authorization succeeded but Gateway account was not found in Connector", errorStatus);
+            return P0199.response();
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java]: is logger.warn good enough for this, I see this as a rare and severe exception to deal with, should be error, or even fatal

File: src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static uk.gov.pay.api.exception.CreateChargeConnectorErrorResponseExceptionMapper.CreateChargeErrorResponse.P0198;
+import static uk.gov.pay.api.exception.CreateChargeConnectorErrorResponseExceptionMapper.CreateChargeErrorResponse.P0199;
+
+public class CreateChargeConnectorErrorResponseExceptionMapper implements ExceptionMapper<CreateChargeConnectorErrorResponseException> {
+
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java]: is it possible to make this mapper class and the exception class more generic one, or we will end having very specific exception classes and the mappers. Also making it more generic will move away from such a long class names

File: src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        LOGGER.info("Error response from Connector unrecognised. status {}, body {}", errorStatus, exception.getErrorBody());
+        return P0198.response();
+    }
+
+    enum CreateChargeErrorResponse {
+
+        P0199(INTERNAL_SERVER_ERROR, "There is an error with this account. Please contact support"),
+        P0198(INTERNAL_SERVER_ERROR, "Downstream system error");
+
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java]: 🎱 👍

File: src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java (hunk 1 above)
Comment (rauligs) [inline src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java]: Will change it to error

File: src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java (hunk 2 above)
Comment (rauligs) [inline src/main/java/uk/gov/pay/api/exception/CreateChargeConnectorErrorResponseExceptionMapper.java]: This story is in progress, If during the refactoring process makes sense will be changed to something more generic but not as part of this PR.

**Pull request merged**